components/cluster/command/upgrade.go

```diff
@@ -49,6 +54,9 @@ func newUpgradeCmd() *cobra.Command {
 			}
 		},
 	}
+	cmd.Flags().StringVarP(&opt.User, "user", "u", utils.CurrentUser(), "The user name to login via SSH. The user must has root (or sudo) privilege.")
```

Comment:
This does not look good to me, adding the requirement of `sudo` user in `upgrade` is:
1. unnecessary, as the deploy user should have sudo permission;
2. introduces new dependency and changes the upgrade procedure of user

Reply:
If opt.User is "" rather than 'root' or other user name, we will get error like ssh handshake fail from https://github.com/pingcap/tiup/blob/7b324055e1b57e46fac715d38afeeef2ee5e6aff/pkg/cluster/manager/manager.go#L240-L242.